tests/integration/_experimental/object_detection/test_evaluator_multiclass_test_case_metrics.py

```diff
@@ -0,0 +1,2086 @@
+                Recall=0.3,
+                F1=0.4285714285714285,
+                AP=0.4166666666666667,
+            ),
+        ],
+        Objects=28,
+        Inferences=15,
+        TP=10,
+        FN=18,
+        FP=5,
+        macro_Precision=0.6875,
+        macro_Recall=0.4,
+        macro_F1=0.4725274725274725,
+        mean_AP=0.5541666666666667,
+    ),
+    (
+        "Threshold: Fixed(0.5) by class, IoU: 0.5, confidence ≥ 0.0",
+        "multiple bboxes test collection",
+    ): TestCaseMetrics(
+        PerClass=[
+            ClassMetricsPerTestCase(
+                Class="a",
+                nImages=7,
+                Threshold=0.5,
+                Objects=14,
+                Inferences=14,
+                TP=9,
+                FN=5,
+                FP=5,
+                Precision=0.6428571428571429,
+                Recall=0.6428571428571429,
+                F1=0.6428571428571429,
+                AP=0.6144957983193277,
+            ),
+            ClassMetricsPerTestCase(
+                Class="b",
+                nImages=6,
+                Threshold=0.5,
+                Objects=8,
+                Inferences=6,
+                TP=4,
+                FN=4,
+                FP=2,
+                Precision=0.6666666666666666,
+                Recall=0.5,
+                F1=0.5714285714285715,
+                AP=0.44642857142857145,
+            ),
+            ClassMetricsPerTestCase(
+                Class="c",
+                nImages=3,
+                Threshold=0.5,
+                Objects=3,
+                Inferences=2,
+                TP=2,
+                FN=1,
+                FP=0,
+                Precision=1.0,
+                Recall=0.6666666666666666,
+                F1=0.8,
+                AP=0.6666666666666666,
+            ),
+            ClassMetricsPerTestCase(
+                Class="d",
+                nImages=1,
+                Threshold=0.5,
+                Objects=1,
+                Inferences=1,
+                TP=1,
+                FN=0,
+                FP=0,
+                Precision=1.0,
+                Recall=1.0,
+                F1=1.0,
+                AP=1.0,
+            ),
+        ],
+        Objects=26,
+        Inferences=23,
+        TP=16,
+        FN=10,
+        FP=7,
+        macro_Precision=0.8273809523809523,
+        macro_Recall=0.7023809523809523,
+        macro_F1=0.7535714285714286,
+        mean_AP=0.6818977591036415,
+    ),
+}
+
+
+@pytest.mark.metrics
+@pytest.mark.parametrize(
+    "config_name, test_name",
+    TEST_PARAMS,
+)
+def test__prebuilt__object__detection__multiclass__compute__test__case__metrics(
+    config_name: str,
+    test_name: str,
+) -> None:
+    config = TEST_CONFIGURATIONS[config_name]
+    eval = ObjectDetectionEvaluator(configurations=[config])
+    random_test_case_name = with_test_prefix("test_evaluator_multiclass") + str(uuid.uuid4())
+    test_case = TestCase(random_test_case_name, reset=True)
```

Comment:
Does this need to create a real test case? This round-trip will slow down these tests significantly — recommend refactoring here to use e.g. a mocked test case, or figuring out a way to make these unit tests instead of integration tests.